src/pages/index.js

```diff
@@ -0,0 +1,368 @@
+}
+
+// Function To Close Modal
+function closeModal(modal) {
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleClickingEscapeKey);
+}
+
+// Function To Close Modal By Clicking On Close Button ("X" Icon) Within Pop Up Form
+function handleClickingCloseButton() {
+  const closeButtons = document.querySelectorAll(".modal__close-button");
+  closeButtons.forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+}
+
+// Function To Close Modal By Clicking Overlay
+function handleClickingOverlay() {
+  modals.forEach((modal) => {
+    modal.addEventListener("click", (event) => {
+      if (event.target === modal) {
+        closeModal(modal);
+      }
+    });
+  });
+}
+
+// Function To Close Modal By Pressing Escape Key ("Esc")
+function handleClickingEscapeKey(event) {
+  if (event.key === "Escape") {
+    const openModal = document.querySelector(".modal_is-opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+// Function For Clicking On Delete Button (Trash Icon)
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteCardModal);
+}
+
+// Function For Clicking On Confirmation Delete Button
+function handleDeleteCardSubmit(event) {
+  event.preventDefault();
+
+  // changes text content to "Deleting..." (setButtonText is method from helpers.js)
+  const deleteButton = event.submitter;
+  setButtonText(deleteButton, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      closeModal(deleteCardModal);
+      selectedCard.remove();
+    })
+    .catch(console.error)
+
+    // changes text content back to "Delete"
+    .finally(() => setButtonText(deleteButton, false, "Delete", "Deleting..."));
+}
+
+// Function For Clicking On Card Image
+function handleCardImageClick(data) {
+  const { _id: id, link, name, isLiked } = data;
+  previewModalImageElement.setAttribute("src", link);
+  previewModalImageElement.setAttribute("alt", name + " image");
+  previewModalCaptionElement.textContent = name;
+  openModal(previewModal);
+}
+
+// Function To Handle Pressing Submit Button For Avatar Modal
+function handleAvatarFormSubmit(event) {
+  event.preventDefault();
+  const saveButton = event.submitter;
+  setButtonText(saveButton, true);
+  api
+    .editUserAvatar(avatarModalLinkInput.value)
+    .then((data) => {
+      avatarImageElement.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(saveButton, false));
+}
+
+// Function To Handle Pressing Submit Button For Edit Profile Modal
+function handleEditProfileFormSubmit(event) {
+  event.preventDefault();
+
+  // changes text content to "Saving..." (setButtonText is method from helpers.js)
+  const saveButton = event.submitter;
+  setButtonText(saveButton, true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions